### PR TITLE
Update auth flow, add signalR reconnects, reactive balances

### PR DESCRIPTION
**Pull request by eli1stark**

### CODE REVIEW

File: lib/service/services/signal_r/service/signal_r_service.dart [hunk 1]
```diff
@@ -1,66 +1,76 @@
...
+import '../model/balance_model.dart';
+import '../model/instruments_model.dart';
+import '../model/prices_model.dart';
+import '../model/server_time_model.dart';
+import 'helpers/signal_r_log.dart';
 
 class SignalRService {
+  SignalRService(this.read);
+
+  final Reader read;
```
Comment (VovaMedvediev) [inline lib/service/services/signal_r/service/signal_r_service.dart:22]: Do you want to override this dependency in tests, right?

File: lib/app/screens/wallet/models/currency_model.dart [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    required double baseBalance,
+  }) = _CurrencyModel;
+
+  const CurrencyModel._();
+
+  bool get isDepositMode => depositMode == 0;
+
+  bool get isWithdrawalMode => withdrawalMode == 0;
+
+  double calculateBaseBalance({
```
Comment (VovaMedvediev) [inline lib/app/screens/wallet/models/currency_model.dart:31]: Is it ok to have business logic in the Model object?

File: lib/app/screens/wallet/providers/currencies_pod.dart [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import '../models/currency_model.dart';
+import 'assets_spod.dart';
+import 'balances_spod.dart';
+import 'converter_map_fpod.dart';
+import 'prices_spod.dart';
+
+final currenciesPod = Provider<List<CurrencyModel>>((ref) {
```
Comment (VovaMedvediev) [inline lib/app/screens/wallet/providers/currencies_pod.dart:9]: Should we extract it to separate function for testing purposes?

File: lib/app/screens/wallet/models/currency_model.dart (hunk 2 above)
Comment (eli1stark) [reply lib/app/screens/wallet/models/currency_model.dart:31]: The model itself is business logic. If logic is related to the model, it makes sense. 
But I'll refactor it anyway to simplify unit testing.

File: lib/service/services/signal_r/service/signal_r_service.dart (hunk 1 above)
Comment (eli1stark) [reply lib/service/services/signal_r/service/signal_r_service.dart:22]: Yeah, I'm sure there is a way. There were some problems with infinite loops that required me to pass explicitly `Reader` that doesn't watch always an object. `Reader` takes a snapshot of the state every time `read(myProvider)` triggers.

File: lib/app/screens/wallet/providers/currencies_pod.dart (hunk 3 above)
Comment (eli1stark) [reply lib/app/screens/wallet/providers/currencies_pod.dart:9]: Will discuss in pm

**Pull request merged**